skactiveml/pool/regression/_information_gain_maximization.py

```diff
@@ -0,0 +1,501 @@
+            the first sample (with index `query_indices[0]`) of the batch.
+            Utilities for labeled samples will be set to np.nan.
+            If candidates is None or of shape (n_candidates), the indexing
+            refers to samples in X.
+            If candidates is of shape (n_candidates, n_features), the indexing
+            refers to samples in candidates.
+        """
+        X, y, candidates, batch_size, return_utilities = self._validate_data(
+            X, y, candidates, batch_size, return_utilities, reset=True
+        )
+
+        check_type(reg, "reg", ProbabilisticRegressor)
+        check_type(fit_reg, "fit_reg", bool)
+        check_type(self.integration_dict, "self.integration_dict", dict)
+        if X_eval is None:
+            X_eval = X
+        else:
+            X_eval = check_array(X_eval)
+            self._check_n_features(X_eval, reset=False)
+
+        X_cand, mapping = self._transform_candidates(candidates, X, y)
+
+        if fit_reg:
+            reg = clone(reg).fit(X, y, sample_weight)
+
+        utilities_cand = self._mutual_information(
+            X_eval, X_cand, mapping, reg, X, y, sample_weight
+        )
+
+        if mapping is None:
+            utilities = utilities_cand
+        else:
+            utilities = np.full(len(X), np.nan)
+            utilities[mapping] = utilities_cand
+
+        return simple_batch(
+            utilities,
+            self.random_state_,
+            batch_size=batch_size,
+            return_utilities=return_utilities,
+        )
+
+    def _mutual_information(
+        self, X_eval, X_cand, mapping, reg, X, y, sample_weight=None
+    ):
+        """Calculates the mutual information gain over the evaluation set if
+        a candidate where to be labeled.
+
+        Parameters
+        ----------
+        X_eval : array-like of shape (n_samples, n_features)
+            The samples where the information gain should be evaluated.
+        X_cand : array-like of shape (n_candidate_samples, n_features)
+            The candidate samples that are potentially labeled.
+        mapping : array-like of shape (n_samples,) or None
+            The potential mapping between the candidate samples and the
+            training data set.
+        reg: ProbabilisticRegressor
+            Predict output values and entropy.
+        X : array-like of shape (n_samples, n_features)
+            Training data set, usually complete, i.e. including the labeled and
+            unlabeled samples.
+        y : array-like of shape (n_samples)
+            Labels of the training data set (possibly including unlabeled ones
+            indicated by `self.missing_label`).
+        sample_weight: array-like of shape (n_samples), optional (default=None)
+            Weights of training samples in `X`.
+
+        Returns
+        -------
+        query_indices : numpy.ndarray of shape (n_candidate_samples)
+            The expected information gain for each candidate sample.
```

Comment:
return value name `query_indices` and the description do not seem to match. Am I missing something?